core/src/main/scala/kafka/server/ActionQueue.scala

```diff
@@ -37,7 +37,10 @@ class ActionQueue {
    * picks up an action to complete.
    */
   def tryCompleteAction(): Unit = {
-    val action = queue.poll()
-    if (action != null) action()
+    var action = queue.poll()
+    while (action != null) {
+      action()
```

Comment:
Main thing to decide is what to do in case of exception, do we stop processing or do we continue?

Reply:
I prefer to just catch it and log a warning as the response has been processed

Reply:
Perhaps we could do the try/catch of each action here instead of KafkaApis. This way, we are guaranteed that all pending actions are processed in time.